File: FibonacciBench/fibonacci.c
// gcc -O3 c.c
#include <stdio.h>

int fibonacci(int n) {
  if (n <= 1) {
    return n;
  } 
  return fibonacci(n - 1) + fibonacci(n - 2);
}

int main(void) {
  printf("%i", fibonacci(40));
}

File: FibonacciBench/fibonacci.zig
// zig build-exe -O ReleaseFast fibonacci.zig
const expect = @import("std").testing.expect;
const std = @import("std");
const print = std.debug.print;

fn fibonacci(index: i64) i64 {
    if (index <= 1) return index;
    return fibonacci(index - 1) + fibonacci(index - 2);
}

pub fn main() !void {
    const x = fibonacci(40);
    print("result = {}\n", .{x});

}